pkg/cluster/cluster.go

```diff
@@ -205,17 +205,18 @@ func (c *Cluster) send(ev *clusterEvent) {
 }
 
 func (c *Cluster) run(stopC <-chan struct{}, wg *sync.WaitGroup) {
-	needDeleteCluster := true
+	clusterFailed := false
 
 	defer func() {
-		if needDeleteCluster {
-			c.logger.Infof("deleting cluster")
+		if clusterFailed {
+			c.reportFailedStatus()
```

Comment:
> On restore, controller will wait all cluster run() exit, using the passed "wg".

this is wrong, and we should fix it. controller should never wait for a specific cluster.